From 4350bd624f165de81de10293a8ec5a59e8b7ce64 Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Fri, 28 Jun 2019 12:22:33 +0200
Subject: [PATCH] Revert "global: combine all permissions of the object tree"

This reverts commit 83bc033837f7525d898f1de91119f669f9bf97f5.

This needs some more work.
---
 src/pipewire/global.c | 10 ++--------
 1 file changed, 2 insertions(+), 8 deletions(-)

diff --git a/src/pipewire/global.c b/src/pipewire/global.c
index c963965e..00258ff8 100644
--- a/src/pipewire/global.c
+++ b/src/pipewire/global.c
@@ -38,15 +38,9 @@ uint32_t pw_global_get_permissions(struct pw_global *global, struct pw_client *c
 {
 	uint32_t perms = PW_PERM_RWX;
 
-	if (client->permission_func == NULL)
-		return perms;
-
-	perms = client->permission_func(global, client, client->permission_data);
-
-	while (global != global->parent) {
-		global = global->parent;
+	if (client->permission_func != NULL)
 		perms &= client->permission_func(global, client, client->permission_data);
-	}
+
 	return perms;
 }
 
